### PR TITLE
feat(contracts-bedrock): StandardBridge inheritance tree (ETHStandardBridge & ERC20StandardBridge)

### PR DESCRIPTION
**Pull request by hamdiallam**
The current StandardBridge base contract natively supports ETH and ERC20 bridging. However for custom implementations, we want users to only depend on the ERC20 portion of the standard bridge as the ETH is a footgun that'll cause locked funds.

Introduce `ETHStandardBridge` and `ERC20StandardBridge` base contracts that mix into the final `StandardBridge`
- Lets have `bridgeERC20` invoke `bridgeERC20To` rather than `_initiateBridgeERC20` such that an implementer is only required to override one function `birdgeERC20To` to cover the both entrypoints  (~50 units of gas increase for bridgeERC20)

`StandardBridgeBase` defines the storage layout in a backwards compatible manner

### CODE REVIEW

**Comment by hamdiallam** [comment]
Current dependencies on/for this PR:
* develop
  * **PR #6673** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6673" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/6673?utm_source=stack-comment).

File: packages/contracts-bedrock/src/L1/L1StandardBridge.sol
Comment (hamdiallam) [inline packages/contracts-bedrock/src/L1/L1StandardBridge.sol]: no-op comment

**Comment by maurelian** [comment]
I applied the `SR-Risk` and `mergify-ignore` labels because this is refactoring mission critical code so we should have multiple code-owner reviews on it before merging.

File: packages/contracts-bedrock/src/universal/StandardBridge.sol
Comment (roninjin10) [inline packages/contracts-bedrock/src/universal/StandardBridge.sol:17]: nit: I would prefer every contract to have it's own file. It makes the artifacts easier to target

**Review by roninjin10 — APPROVED** [review]
lgtm will need a lot more contract reviewers though

**Pull request closed without merging**
